88.md

```diff
@@ -0,0 +1,195 @@
+NIP-88
+======
+
+Nostr Cash (simple Nostr cash/token/cheque)
+-----------------------------------
+
+`draft` `optional` `author:benarc`
+
+Electronic cash is useful and relays should have the option to offer minting/custodial services. Other mints could exist as stand-alone clients, but relays are ideally placed to package in with their service offerings.
+
+Users can use relays offering mints they trust, or relays they don't trust but for smaller amounts. 
+
+Relays offering mints must have a keypair and an additional nostr keypair for its mint.
+
+Nostr Cash uses derived keys, much like [NIP26](https://github.com/nostr-protocol/nips/blob/master/26.md). Each "cash" is actually just a derived private key that can be burned or passed to another user.
+
+All communication between the relay and user happen over [NIP04](https://github.com/nostr-protocol/nips/blob/master/04.md).
+
+The mints ledger is two tables `active` and `spent`:
+
+```
+active:
+    key: <integer, keypair deriviation number from master>, 
+    amount: <integer BTC millisat denomination>, 
+    user: <string, public-key of user>,
+    timestamp: <integer timestamp>
+
+spent:
+    key: <integer, keypair deriviation number from master>, 
+    amount: <integer BTC millisat denomination>, 
+    user: <string, public-key of user>,
+    timestamp: <integer timestamp>
+```
+
+User client cash wallet stores:
+
+```
+active:
+    key: <string, derived private key/cash>, 
+    amount: <integer BTC millisat denomination>, 
+    relay: <string, relay ws address>,
+    relaykey: <string, public-key of relay>,
+    timestamp: <integer timestamp>
+
+spent:
+    key: <string, derived private key/cash>, 
+    amount: <integer BTC millisat denomination>, 
+    relay: <string, relay ws address>,
+    relaykey: <string, public-key of relay>,
+    timestamp: <integer timestamp> 
+```
+
+## Workflows
+
+### Mint details 
+
+**For fetching info about the mint.**
+
+User DMs the relay:
+
+```json
+{
+    "mint": <bool true>
+}
+```
+
+Relay DMs user (maybe after a check if the relays mint has restricted access):
+
+```json
+{
+    "title": <string mints title>,
+    "description": <string mints description>,
+    "liquiity": <integer optional public millisat balance of mint>,
+    "type": <list different offered payment types bolt11 bolt12 onchain stripe paypal shitcoin etc>,
+    "fee": <integer millisat fee for internal nostr payments>,
+}
+```
+
+### Minting
+
+**For creating new cash from a mint.**
+
+User DMs the relay:
+
+```json
+{
+  "amount": <integer millisats>
+}
+```
+
+Relay responds with payment request, which user pays:
```

Comment:
Should use [NIP-15 payment_options](https://github.com/nostr-protocol/nips/blob/master/15.md#step-2-merchant-request-payment-event) for consistency

Reply:
Yes! Great